web_src/js/modules/tippy.js

```diff
@@ -86,29 +97,32 @@ function attachChildrenLazyTooltip(target) {
   }
 }
 
+const elementNodeTypes = new Set([Node.ELEMENT_NODE, Node.DOCUMENT_FRAGMENT_NODE]);
+
 export function initGlobalTooltips() {
-  // use MutationObserver to detect new elements added to the DOM, or attributes changed
-  const observer = new MutationObserver((mutationList) => {
+  // use MutationObserver to detect new "data-tooltip-content" elements added to the DOM, or attributes changed
+  const observerConnect = (observer) => observer.observe(document, {
+    subtree: true,
+    childList: true,
+    attributeFilter: ['data-tooltip-content', 'title']
+  });
+  const observer = new MutationObserver((mutationList, observer) => {
+    observer.disconnect();
```

Comment:
Why disconnect/re-observe?

Reply:
To improve performance slightly (in case anything in DOM is changed in our callback)

Reply:
Hmm do you think this would prevent infinite loops if the element itself changes which would have re-triggered the observer?

Reply:
Perhaps we should add a comment for that?

Reply:
IMO since we also check the "title is empty" in the callback, so even without the `disconnect`, there won't be infinite loop. The logic could be stable enough.

Reply:
The time element has a special behaviour where it won't touch title if it is empty string, but other elements may not have such a rule.

Reply:
I think we should remove the unregistering. I don't think browsers can guarantee that these mutation events come serially, so if an mutation comes in while the current callback is still running, it will be missed.

Reply:
I think your point is right, according to MDN: [takeRecords()](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/takeRecords)

We need to do `takeRecords` after disconnect.

Reply:
At this point, I'm wondering if it's even worth the effort to disconnect. It'll make the code much harder to understand for questionable benefit.

Reply:
> It guarantees. JavaScript is a single thread program, nothing is concurrent.

It's not true concurrency (e.g. multiple threads), but parallelism via event loop can still happen.

Reply:
`async` is just a syntax sugar. If you expand it to the legacy Promise syntax, you will see that nothing different in the end. Event loop also guarantees serialization if our code doesn't exit from our function.